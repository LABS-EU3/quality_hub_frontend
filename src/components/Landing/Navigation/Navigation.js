import React from 'react';
import { Link } from 'react-router-dom';
import { Logo } from '../Landing-styles';
import { NavContainer } from './NavigationStyles';

const Navigation = () => (
  <NavContainer data-testid='landing-nav' className='navigation'>
    <div className='logo'>
      <Link to='/'>
        <Logo />
      </Link>
      <p className='logo-name'>DevCoach</p>
    </div>
    <div className='list-items-container'>
      <ul className='list-items'>
        <li className='list-item'>
          <Link data-testid='faq' to='/faq/general'>
            FAQ
          </Link>
        </li>
        <li className='list-item'>
          <Link data-testid='about' to='/about'>
            About
          </Link>
        </li>
        <li className='list-item'>
          <Link data-testid='login' to='/login'>
            Login
          </Link>
        </li>
        <li className='list-item'>
          <Link data-testid='signup' to='/register'>
            Signup
          </Link>
        </li>
      </ul>
    </div>
  </NavContainer>
);

export default Navigation;
