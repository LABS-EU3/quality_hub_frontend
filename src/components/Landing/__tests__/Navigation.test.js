import React from 'react';
import ReactDOM from 'react-dom';
import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter as Router } from 'react-router-dom';
import * as rtl from '@testing-library/react';
import renderer from 'react-test-renderer';
import Navigation from '../Navigation/Navigation';

describe('Navigation', () => {
  it('matches snapshot ', () => {
    const tree = renderer
      .create(
        <Router>
          <Navigation />
        </Router>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Router>
        <Navigation />
      </Router>,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Signup link ok', () => {
    const tree = rtl.render(
      <Router>
        <Navigation />
      </Router>,
    );
    const signup = tree.getByTestId('signup');
    expect(signup).toBeInTheDocument();

    expect(signup.closest('a')).toHaveAttribute('href', '/register');
  });

  it('FAQ Link ok', () => {
    const tree = rtl.render(
      <Router>
        <Navigation />
      </Router>,
    );
    const faq = tree.getByTestId('faq');
    expect(faq).toBeInTheDocument();

    expect(faq.closest('a')).toHaveAttribute('href', '/faq/general');
  });

  it('About link ok', () => {
    const tree = rtl.render(
      <Router>
        <Navigation />
      </Router>,
    );
    const about = tree.getByTestId('about');
    expect(about).toBeInTheDocument();

    expect(about.closest('a')).toHaveAttribute('href', '/about');
  });

  it('Login link ok', () => {
    const tree = rtl.render(
      <Router>
        <Navigation />
      </Router>,
    );
    const login = tree.getByTestId('login');
    expect(login).toBeInTheDocument();

    expect(login.closest('a')).toHaveAttribute('href', '/login');
  });
});
