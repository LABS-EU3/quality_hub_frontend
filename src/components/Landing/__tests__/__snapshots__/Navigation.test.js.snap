// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation matches snapshot  1`] = `
<nav
  className="sc-bxivhb frcxEG navigation"
  data-testid="landing-nav"
>
  <div
    className="logo"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <div
        className="sc-bdVaJa kHYcQv"
      />
    </a>
    <p
      className="logo-name"
    >
      DevCoach
    </p>
  </div>
  <div
    className="list-items-container"
  >
    <ul
      className="list-items"
    >
      <li
        className="list-item"
      >
        <a
          data-testid="faq"
          href="/faq/general"
          onClick={[Function]}
        >
          FAQ
        </a>
      </li>
      <li
        className="list-item"
      >
        <a
          data-testid="about"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </li>
      <li
        className="list-item"
      >
        <a
          data-testid="login"
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </li>
      <li
        className="list-item"
      >
        <a
          data-testid="signup"
          href="/register"
          onClick={[Function]}
        >
          Signup
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
